yjit_codegen.c

```diff
@@ -3488,11 +3488,8 @@ gen_send_iseq(jitstate_t *jit, ctx_t *ctx, const struct rb_callinfo *ci, const r
         }
 
         if (vm_ci_flag(ci) & VM_CALL_KWARG) {
-            if ((kw_arg->keyword_len != keyword->num) || (lead_num != argc - kw_arg->keyword_len)) {
-                // Here the method being called specifies optional and required
-                // keyword arguments and the callee is not specifying every one
-                // of them.
-                GEN_COUNTER_INC(cb, send_iseq_kwargs_req_and_opt_missing);
+            if (lead_num != argc - kw_arg->keyword_len) {
+                GEN_COUNTER_INC(cb, send_iseq_complex_callee);
```

Comment:
Would be good to have a comment here to explain what this if statement is checking

As someone not familiar with this code, this is cryptic: `lead_num != argc - kw_arg->keyword_len`

Reply:
This is checking that the number of non-keyword arguments match. Added a comment.

I think soon we'll be able to handle both optional positional and keyword args in the same call and this check will change then.